### ISSUE TITLE
Create a command line option to pick the correct/verified chain in a rollback scenario

### ISSUE THREAD

**Issue by mms710** [zcash/zcash#3853]
As part of our commitment to rollback protection, we need to create a command line option that allows people to pick the correct/verified chain in a rollback scenario

**Comment by tromer** [comment]
So this command-line option will override the "heaviest valid chain" consensus rule?

Put otherwise, the consensus rule becomes "heaviest valid chain that includes the block specified in the command line"?

So if the recommended chain fork hasn't yet accumulated more PoW weight than the not-recommended one, and someone runs `zcashd` without the special command line, they'll end up on the not-recommended chain fork?

**Comment by zebambam** [comment]
That's right. It's a ticket that describes a simple way to enact a social consensus on what is the valid chain, specifically overriding the one with the most work, making this rule more important than the one about the chain containing the most work wins.

**Comment by zebambam** [comment]
I agree though - just as booting nodes have to be seeded onto the network, they would have to be "seeded" with the latest consensus so as not to leave them behind. So the ticket is probably more than just a command-line option, but it's a good straw-person approach to argue against.

**Comment by tromer** [comment]
The generic answer is "new release of the zcashd (and compatible) software"; with the command-line option serving just as an interim measure while the new release is cooked.

**Comment by zebambam** [comment]
Yeah that's been the statement given out when people have asked about this in the past. This is an attempt to make that approach more slick, reduce workload on our releases and decrease the time needed to get up and running again.
